### PR TITLE
Add a pattern to the git BranchQuery

### PR DESCRIPTION
**Pull request by alexr00**
It turns out that this:
```
git for-each-ref --format "%(refname) %(objectname)" --contains 4ea4be72675c61c1eb62caa36377df065a1cdd68
```
is impressively slow in repositories with many branches (I saw >15 seconds, which is unusable for what I was using it for). However, this:
```
git for-each-ref --format "%(refname) %(objectname)" refs/remotes/origin/HEAD --contains 4ea4be72675c61c1eb62caa36377df065a1cdd68
```
is much quicker (<4 seconds).

To accomplish this, I've added a pattern (which is what that arg is called in the [docs](https://git-scm.com/docs/git-for-each-ref)) to the API.

### CODE REVIEW

**Comment by joaomoreno** [comment]
Interesting, in which feature are you actually calling this?

**Comment by alexr00** [comment]
This will replace the way I was using `getBranch` in the GHPRI extension to determine the best branch to create a permalink for. 
The old way was to get the world using `getBranch` for a commit then do filtering and sorting based on the "best" remotes and branches. For microsoft/vscode, this always takes >10 seconds.
The new way would be to make the "best" patterns (remote+branch combos) then try to `getBranch` for each of these patterns. Best case, the first pattern returns a successful branch, which should only be a few (I think I've seen 2 or 3 seconds max) seconds. Worst case, we test all patterns (likely less than 6).

**Comment by joaomoreno** [comment]
Yeah I noticed it was indeed slower! 👍

File: extensions/git/src/git.ts [hunk 1]
```diff
@@ -1791,13 +1791,21 @@ export class Repository {
...
+			args.push(opts.pattern);
+		}
+
+		if (opts?.count) {
+			args.push(`--count=${opts.count}`);
+		}
+
 		if (opts?.contains) {
 			args.push('--contains', opts.contains);
 		}
```
Comment (joaomoreno) [inline extensions/git/src/git.ts:1811]: Let's make sure the order or args matches the order in the docs: https://git-scm.com/docs/git-for-each-ref#_synopsis

**Review by joaomoreno — APPROVED** [review]
Apart from the order of args, everything else looks great, feel free to merge once fixed.

File: extensions/git/src/git.ts (hunk 1 above)
Comment (alexr00) [reply extensions/git/src/git.ts]: Updated order of for-each-ref args to be count, sort, format, pattern, then contains.

**Pull request merged**